feat(tests): ✨ add layer policy tests and helper functions OC:6729

carlacupani opened the pull request:
- Introduced `LayerTestHelpers.php` to encapsulate helper functions for creating users and layers in tests.
  - `createUserWithRole`: Creates a user and assigns a role, with special handling for 'regular' users.
  - `createLayer`: Creates a layer with an optional user ID.

- Implemented `LayerPolicyTest.php` to verify various user roles' abilities to interact with layers.
  - Tests cover administrator and validator roles.
  - Ensured administrators can create, update, and delete any layer.
  - Validated that validators cannot create, update, or delete layers but can view them.

carlacupani committed:
feat(tests): ✨ add Layer policy tests with helper functions OC:6729

- Introduced `LayerTestHelpers.php` to provide helper functions `createUserWithRole` and `createLayer` for test setup.
- Added `LayerPolicyTest.php` to verify `update`, `delete`, `create`, and `view` abilities for `Administrator` and `Validator` roles.
- Utilized Laravel's `Gate` facade to assert permissions based on user roles.
- Ensured comprehensive coverage of policy rules for `Layer` model with `DatabaseTransactions` to maintain test isolation.

peppedeka commented on tests/Feature/Helpers/LayerTestHelpers.php:
```diff
@@ -0,0 +1,23 @@
+    function createUserWithRole(string $role): User
+    {
+        $user = User::factory()->create();
+        if ($role !== 'regular') {
```
esiste un ruolo regular?

carlacupani committed:
refactor(tests): 🔧 convert LayerPolicyTest to use trait for setup and assertions OC:6729

- Moved the `createUserWithRole` and `createLayer` functions into a `LayerTestHelpers` trait for reusability and maintainability.
- Refactored `LayerPolicyTest` to use PHPUnit's `TestCase` class and incorporated the `LayerTestHelpers` trait to streamline test setup and improve readability.
- Expanded test coverage by implementing detailed permissions tests for different user roles:
  - Added comprehensive tests for 'Administrator', 'Validator', and users without specific roles.
  - Ensured all necessary tests verify permissions for viewing, creating, updating, and deleting 'Layer' entities.
